### PR TITLE
[CRIMAPP-835] Add IoJ answer validator

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change

Validate Interests of Justice answers on Submission Review

## Link to relevant ticket

https://dsdmoj.atlassian.net/browse/CRIMAPP-835

## Notes for reviewer

Adds an answer validator for the IoJ task and adds it to the section completeness validator.

## How to manually test the feature

Confirm that an error message is shown when IoJ are required but have not been given (or have only been partially given). See ticket for details.

### CODE REVIEW

File: app/validators/interests_of_justice/answers_validator.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    def validate
+      return if complete?
+
+      errors.add(:ioj, :incomplete)
+      errors.add(:base, :incomplete_records)
+    end
+
+    def complete?
+      crime_application.ioj_passported? || ioj_answers_complete?
```
Comment (arthurashman) [inline app/validators/interests_of_justice/answers_validator.rb:23]: Wondering whether `passported` is more of a not-applicable scenario - could it confuse a user to update to `complete` when they did not complete anything?

Comment (timpeat) [reply app/validators/interests_of_justice/answers_validator.rb:23]: Ideally, we want the task list's 'non-applicable' and the answer validator's 'applicable?' to be inverses of each other. For this to happen, 'applicable?' in the validator should function as the inverse of 'not_applicable?'.

For example, in a "change in financial circumstances" application, the 'Interests of Justice' section is not applicable. However, for an under-18 applicant, the IoJ is applicable to the application but passported, so it is automatically completed.

I think we're at the point where we can start defining  what 'applicable?' means in the context of an answer validator, and we now have enough of a pattern to consider a refactor and see if we can standardise each interface and clearly define their roles. Does that make sense?

File: app/presenters/tasks/ioj.rb [hunk 2]
```diff
@@ -8,29 +8,18 @@ def path
...
     def can_start?
       fulfilled?(CaseDetails)
     end
 
     def in_progress?
-      crime_application.ioj_passport.any? || ioj.present?
+      ioj.present?
     end
 
-    def completed?
```
Comment (arthurashman) [inline app/presenters/tasks/ioj.rb:25]: I think removing this has meant that the tasklist badge doesn't show complete anymore

File: app/validators/interests_of_justice/answers_validator.rb (hunk 1 above)
Comment (arthurashman) [reply app/validators/interests_of_justice/answers_validator.rb:23]: Yep, that makes sense.

**Pull request merged**